kyuubi-common/src/main/scala/org/apache/kyuubi/util/ThreadUtils.scala

```diff
@@ -95,4 +96,53 @@ object ThreadUtils extends Logging {
       }
     }
   }
+
+  def runInNewThread[T](
+      threadName: String,
+      isDaemon: Boolean = true)(body: => T): T = {
+    @volatile var exception: Option[Throwable] = None
+    @volatile var result: T = null.asInstanceOf[T]
+
+    val thread = new Thread(threadName) {
+      override def run(): Unit = {
+        try {
+          result = body
+        } catch {
+          case NonFatal(e) =>
+            exception = Some(e)
+        }
+      }
+    }
+    thread.setDaemon(isDaemon)
+    thread.start()
+    thread.join()
```

Comment:
The join method here will cause the engine to get stuck.

Reply:
Yes, that's ture. Thank you for pointing out.
Shall I change the name of `runInNewThread` as well? (So that others may notice the difference with the runInNewThread in Spark.)